feat(core): extract platforms out of core

vsavkin opened the pull request:
Currently, core depends on the browser, which means that other platforms (e.g., NativeScript or webworker) cannot use the bootstrapping logic core provides.
This PR extract makes bootstrapping logic in core completely platform-independent. The browser-specific code was moved to "angular2/platforms/browser".

BREAKING CHANGE

A few private helpers (e.g., platformCommon or applicationCommon) were removed or replaced with other helpers. Look at PLATFORM_COMMON_PROVIDERS, APPLICATION_COMMON_PROVIDERS, BROWSER_PROVIDERS, BROWSER_APP_PROVIDERS to see if they export the providers you need.

IgorMinar commented on modules/angular2/test/public_api_spec.ts:
```diff
@@ -126,6 +127,8 @@ var NG_ALL = [
   'Binding.toFactory',
   'Binding.toValue',
+  'BROWSER_APP_PROVIDERS:js',
+  'BROWSER_PROVIDERS:js',
```
js?

mary-poppins commented:
Merging PR #5219 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5219](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5219).

mary-poppins commented:
Merging PR #5219 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5219](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5219).

mary-poppins commented:
Merging PR #5219 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5219](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5219).

vsavkin committed:
feat(core): extract platforms out of core

Currently, core depends on the browser, which means that other platforms (e.g., NativeScript or webworker) cannot use the bootstrapping logic core provides.
This PR extract makes bootstrapping logic in core completely platform-independent. The browser-specific code was moved to "angular2/platforms/browser".

BREAKING CHANGE

A few private helpers (e.g., platformCommon or applicationCommon) were removed or replaced with other helpers. Look at PLATFORM_COMMON_PROVIDERS, APPLICATION_COMMON_PROVIDERS, BROWSER_PROVIDERS, BROWSER_APP_PROVIDERS to see if they export the providers you need.

mary-poppins commented:
Merging PR #5219 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5219](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5219).
